benchexec/tools/sv_benchmarks_util.py

```diff
@@ -40,3 +44,195 @@ def get_data_model_from_task(task, param_dict):
+    )
+    return other_files
+
+
+def get_non_witness_input_files(task):
+    """
+    This function returns the non-witness input files from the task.
+    They consist of all files which do not match the witness file name.
+    The witness file is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: List of non-witness files
+    """
+    _, other_files = _partition_input_files(task.input_files, task.options)
+    if not other_files:
+        raise UnsupportedFeatureException(
+            "Tool does not support tasks without input files"
+        )
+    return other_files
+
+
+def get_single_non_witness_input_file(task):
+    """
+    This function returns the single non-witness file from the task.
+    It raises an exception if there are multiple non-witness files.
+    Non-witness files consist of all files which do not match the witness file name.
+    The witness file is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: Single non-witness file
+    """
+    other_files = get_non_witness_input_files(task)
+    if len(other_files) > 1:
+        raise UnsupportedFeatureException(
+            "Tool does not support tasks with more than one input file"
+        )
+    return other_files[0]
+
+
+def get_witness_options(options, task, witness_options):
+    """
+    This function returns the additional options to handle the witness.
+    It checks if the witness is passed as an option or through the task definition.
+    If the witness is passed through both, it raises an exception.
+    If no witness is given in the task, it returns an empty list.
+    Therefore, this function can be used with any task regardless if it
+    has a witness option or not.
+
+    @param options: List of existing options
+    @param task: An instance of a task
+    @@param witness_options: List of options which need to be set to handle the witness
+        e.g. if the option is "-w witness_file", then the witness should be passed as
+        ["-w", "witness_file"] to the tool and the witness_options parameter is ["-w"]
+
+    @return: List of additional options to handle the witness. They are constructed
+        based on witness_options. For example if witness_options is ["-w"], then
+        the return value will be ["-w", "witness_file"]
+    """
+
+    if isinstance(task.options, dict) and "witness" in task.options.keys():
+        if not any(witness_option in options for witness_option in witness_options):
+            additional_options = []
+            for witness_option in witness_options:
+                additional_options += [witness_option, get_witness(task)]
+            return additional_options
+        else:
+            logging.warning(
+                "You are passing a witness as both an option and through the task definition. "
+                "Using the witness passed as option."
```

Comment:
Maybe something stronger, like "The witness in the task definition will be ignored."?

Reply:
Tried to improve the formulation in 3dd7093997a752a253ca792fd65c88fc0276c110, hopefully that works.